[WFCORE-7003][WFCORE-7007] In testsuite, do not allow Bootable JAR to start in adm…

yersan opened the pull request:
…in-only or suspend modes

Jira issue: https://issues.redhat.com/browse/WFCORE-7003
Jira issue: https://issues.redhat.com/browse/WFCORE-7007

yersan commented:
@bstansberry yes, there are some need to be fixed indeed

bstansberry commented:
In full WF the only use of these methods is one call in org.wildfly.test.manual.management.ManagementOnlyModeTestCase, and that module doesn't run in bootable jar test profiles. So this should be fine there.

I note this because CI for this PR wouldn't catch any problems in full WF as it doesn't exercise bootable jar there.

jfdenise commented on testsuite/test-runner/src/main/java/org/wildfly/core/testrunner/Server.java:
```diff
@@ -187,6 +187,9 @@ protected void start(PrintStream out) {
             CommandBuilder cbuilder = null;
             boolean needNormalModeCheck = false;
             if (isBootableJar) {
+                if (this.startMode != StartMode.NORMAL) {
```
@bstansberry , looking at how tests evolved, it appears that some tests start the bootable JAR in adminMode and succeed (no restart of the server). I ran some tests that we are excluding (because of adminMode) and they are passing. I think that the reasoning was: we are not exposing the option to start a bootable JAR in admin mode, so we excluded tests that use admin mode although they could succeed.

yersan replied:
@jfdenise I am a bit lost because Brian's question was more about starting them in SUSPEND. 

So, the problem with the Bootable JAR is not about whether it can be started in admin-only (or suspend), it is more that we do not "expose" those options? What does "expose" mean here? are we talking about it is not documented as a valid option? .... my very naive understanding is that changes are not persisted, so if you start in admin-only to later move to normal, as soon as you restart instead of reload, your changes will be lost. In such a case, I guess starting in SUSPEND to go later to NORMAL could be valid as long as you do not modify the server configuration

bstansberry replied:
@jfdenise Unless you edited the class we're discussing here to actually pass through the --admin-only to the bootable jar process, enabling more tests (I assume by removing pom.xml exclude elements) with the existing code basically demonstrates that the test didn't need to use startInAdminMode at all. That was definitely the case with ManagementInterfaceResourcesTestCase.

So perhaps before merging this we should enable a lot of tests and see which pass. If they pass, the test doesn't need StartMode.ADMIN_ONLY and we can can fix them to not use it and leave them enabled.

bstansberry replied:
@jfdenise You didn't answer my question about StartMode.SUSPEND though. :)

jfdenise replied:
@bstansberry , for SUSPEND I would say that we should allow it and set the mode. I don't see why a Bootable JAR couldn't be started in SUSPEND. So perhaps this fix is too radical.

yersan replied:
> So perhaps this fix is too radical

@jfdenise @bstansberry In any case, since Bootable JAR does not expose the ability to start in any other mode that is not `normal`, we have to continue rejecting both `suspend` and `admin-only`, otherwise, the test would assume a condition that will never be satisfied by the Bootable JAR.

I've reviewed the current set (org.jboss.as.test.manualmode.suspend.*) that requires suspend mode, and they cannot be changed since suspend is a hard requirement for those tests. I've just switched one test that can be run in normal mode instead of admin-only.

bstansberry replied:
> 
> @jfdenise @bstansberry In any case, since Bootable JAR does not expose the ability to start in any other mode that is not `normal`, we have to continue rejecting both `suspend` and `admin-only`, otherwise, the test would assume a condition that will never be satisfied by the Bootable JAR.

Makes sense; thanks.

> 
> I've reviewed the current set (org.jboss.as.test.manualmode.suspend.*) that requires suspend mode, and they cannot be changed since suspend is a hard requirement for those tests. I've just switched one test that can be run in normal mode instead of admin-only.

Thank you!

bstansberry commented on testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/management/persistence/yaml/YamlExtensionTestCase.java:
```diff
@@ -330,7 +331,7 @@ public void testSimpleYamlWithCliBootOps() throws Exception {
     @Test
     public void testYamlChangesAppliedInAdminOnlyModeWithoutBootCliScript() throws Exception {
-        container.start(null, null, Server.StartMode.ADMIN_ONLY, System.out, false, null, null, null, null, new Path[]{testYaml});
+        container.start(null, null, Server.StartMode.NORMAL, System.out, false, null, null, null, null, new Path[]{testYaml});
```
This is contrary to what the test method name says is being tested.

yersan replied:
Upss, so evident. Fixed, move it back to ADMIN_ONLY to make honor test intention.

yersan commented:
It didn't go well, taking a closer look

yersan commented:
@bstansberry  The `ShutdownTestCase` affected the test executions, after adding it back to the exclusion list, it went all back to normal. I've squashed and rebased the commits.

bstansberry commented:
Thanks @yersan and @jfdenise
